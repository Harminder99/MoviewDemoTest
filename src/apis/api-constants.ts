export const ImageBaseUrl = 'https://image.tmdb.org/t/p/w300/';
export const ApiKeyMovies = 'Your -- Key -- Here';
export const PopularMovies = `/3/movie/popular?api_key=${ApiKeyMovies}`;
