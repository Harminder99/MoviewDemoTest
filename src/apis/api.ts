// api.js
// Import the Axios library for making HTTP requests

import axios from 'axios';

// Create an Axios instance called 'api' with a base URL for your API
const api = axios.create({
  baseURL: 'https://api.themoviedb.org', // Replace with your API base URL
});

// Export the 'api' instance to make it available for use in other parts of your code
export default api;
