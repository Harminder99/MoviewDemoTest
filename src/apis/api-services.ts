import {Movies} from './api-types';
import {PopularMovies} from './api-constants';
import api from './api';
import {useQuery} from '@tanstack/react-query';

// this hook will help you get response from Api
export function useMoviesList() {
  return useQuery<Movies, Error>(['movie'], async () => {
    const {data} = await api.get(PopularMovies);
    return data;
  });
}
