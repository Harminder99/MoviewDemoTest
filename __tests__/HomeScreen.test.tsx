import {fireEvent, render, waitFor} from '@testing-library/react-native';

import HomeScreen from '../src/ui/home/home-screen'; // Replace with your actual path
import { Movie } from '../src/apis/api-types';
import React from 'react';
import { useMoviesList } from '../src/apis/api-services';

// Mock a sample movie data for testing
const movieList:Array<Movie> = [{
  title: 'Test Movie Title',
  overview: 'Test Movie Description',
  poster_path: '/test-poster.jpg',
  vote_count: 100,
  adult: true,
  backdrop_path: '',
  genre_ids: [],
  id: 101,
  original_language: '',
  original_title: '',
  popularity: 0,
  release_date: '',
  video: false,
  vote_average: 0,
}];

// Mock your dependencies as needed, such as useDispatch and useNavigation
jest.mock('react-redux', () => ({
  useDispatch: () => jest.fn(),
}));
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));
jest.mock('../src/apis/api-services.ts', () => ({
  useMoviesList: jest.fn(),
}));

describe('HomeScreen', () => {

  it('renders correctly', () => {
    const tree = render(<HomeScreen />);
    expect(tree).toMatchSnapshot();
  });

  it('renders error message when there is an error', async () => {
  // Mock the useMoviesList hook to return an error
    (useMoviesList as jest.Mock).mockReturnValue({error: new Error('Test error')})

    const {getByTestId} = render(<HomeScreen />);
    await waitFor(() => expect(getByTestId('error-message')).toBeTruthy());
  });

  it('renders empty state message when there is no data', async () => {
    // Mock the useMoviesList hook to return no data
    (useMoviesList as jest.Mock).mockReturnValue({data: {results: []}, isLoading: false})

    const {getByTestId} = render(<HomeScreen />);
    await waitFor(() => expect(getByTestId('no-movie-message')).toBeTruthy());
  });

  it('should render list of 1 movie item', async () => {
    // Mock the useMoviesList hook to return no data
    (useMoviesList as jest.Mock).mockReturnValue({data: {results: movieList}, isLoading: false})

    const {getByTestId} = render(<HomeScreen />);
    await waitFor(() => expect(getByTestId('movie-list').children).toHaveLength(movieList.length));
  });
});
