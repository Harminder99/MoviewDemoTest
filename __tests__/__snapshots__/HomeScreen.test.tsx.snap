// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    data={Array []}
    getItem={[Function]}
    getItemCount={[Function]}
    keyExtractor={[Function]}
    onContentSizeChange={[Function]}
    onLayout={[Function]}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    showsHorizontalScrollIndicator={false}
    showsVerticalScrollIndicator={false}
    stickyHeaderIndices={Array []}
    testID="movie-list"
    viewabilityConfigCallbackPairs={Array []}
  >
    <View />
  </RCTScrollView>
</View>
`;
